src/compiler/checker.ts

```diff
@@ -24699,6 +24700,20 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
         }
 
+        function inferFromMappedProperties(source: Type, target: MappedType) {
+            const properties = getPropertiesOfObjectType(source);
+            for (const sourceProp of properties) {
+                if (!some(sourceProp.declarations, hasSkipDirectInferenceFlag)) {
+                    const sourcePropType = getTypeOfSymbol(sourceProp);
+                    if (!couldContainTypeVariables(sourcePropType)) {
+                        continue;
+                    }
+                    const targetPropType = substituteIndexedMappedType(target, getStringLiteralType(unescapeLeadingUnderscores(sourceProp.escapedName)));
+                    inferFromTypes(sourcePropType, targetPropType);

```

Comment:
Creating the literal property type like this seems sketchy to me. Maybe you should use `getLiteralTypeFromProperty` instead?

Reply:
I used the method of creating a named~ property type from here:
https://github.dev/microsoft/TypeScript/blob/7e4592ea096b626274e7b20550f2f9cbad9a162f/src/compiler/checker.ts#L28960-L28962

I can adjust this later if needed but it would be cool to know what's the difference between both methods.